From 8d2ef1b1a5b04609f61b40047057bd44c87b2511 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 25 Jul 2024 06:16:02 +0300
Subject: [PATCH] [FSW-2704] fs1: boot secondary ppboot

This patch adds booting secondary ppboot in case
primary ppboot did not boot. If secondary ppboot
also does not boot then rom will execute
recvoery.

Change-Id: I6c7bdb63da59657a03c02bc8cebc74df124936ee
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index cca969f..8cd5574 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -80,5 +80,17 @@
 		boot_rom_halt();
 	}
 
+	boot_rom_boot_flow_wait_for_dma_finish();
+
+	/* If the control reaches here start prefetch of recovery image */
+	boot_rom_boot_flow_prefetch_img(HL_FW_IMG_RECOVERY);
+
+	if (boot_rom_boot_flow_is_ppboot_valid(BOOT_ROM_SECONDARY)) {
+		boot_rom_boot_flow_boot_ppboot_sub(BOOT_ROM_SECONDARY);
+		boot_rom_halt();
+	}
+
+	execute_recovery();
+
 	return 0;
 }
